File: LeetCode/Medium/0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.ts
function characterReplacement(s: string, k: number): number {
    let left = 0;
    let right = 0;
    let result = 0;
    let map = new Map<string, number>();

    while (right < s.length) {
        map.set(s[right], (map.get(s[right]) || 0) + 1);

        const maxFreq = Math.max(...Array.from(map.values()));
        if ((right - left + 1) - maxFreq > k) {
            map.set(s[left], map.get(s[left])! - 1);
            left++;
        }

        result = Math.max(result, right - left + 1);
        right++;
    }

    return result;
}

File: LeetCode/Medium/0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        count = {}
        start = 0
        max_count = 0
        result = 0
        for i in range(len(s)):
            count[s[i]] = count.get(s[i], 0) + 1
            max_count = max(max_count, count[s[i]])
            while (i - start + 1) - max_count > k:
                count[s[start]] -= 1
                start += 1
            result = max(result, i-start + 1)
        
        return result

            

        
            
